projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/credentials/AWSCredentialsService.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.credentials;
+
+import com.amazonaws.auth.AWSSessionCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest;
+import com.vmware.taurus.exception.AuthorizationError;
+import java.util.Map;
+import java.util.UUID;
+import lombok.Getter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * This Spring Service Component provides a convenient way to manage AWS SDK credentials within
+ * control-service. It simplifies the process of setting up and configuring AWS SDK credentials,
+ * allowing you to quickly get up and running with AWS temporary session credentials for service
+ * accounts with the aws credentials found in the application.properties file. More information
+ * about the properties and service user patterns can be found in the application.properties file's
+ * comments or https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
+ */
+@Service
+public class AWSCredentialsService {
+
+  public static final String AWS_SERVICE_ACCOUNT_SECRET_ACCESS_KEY =
+      "AWS_SERVICE_ACCOUNT_SECRET_ACCESS_KEY";
+  public static final String AWS_SERVICE_ACCOUNT_ACCESS_KEY_ID =
+      "AWS_SERVICE_ACCOUNT_ACCESS_KEY_ID";
+  public static final String AWS_SERVICE_ACCOUNT_SESSION_TOKEN =
+      "AWS_SERVICE_ACCOUNT_SESSION_TOKEN";
+
+  @Getter private String awsRegion;
+
+  @Getter private boolean assumeIAMRole;
+
+  @Getter private String awsSecretAccessKey;
+
+  @Getter private String awsAccessKeyId;
+
+  private STSAssumeRoleSessionCredentialsProvider credentialsProvider;
+
+  public AWSCredentialsService(
```

Comment:
I think we need documentation for the method.

Particularly what is the difference between awsSecretAccessKey and serviceAccountSecretKey and such. 

Also, 10 arguments for method /constructor is a lot. 
Maybe use [ConfigutationProperties](https://www.baeldung.com/configuration-properties-in-spring-boot)

It should be something like 

```
@ConfigutationProperties("datajobs.aws")
AWSCredentialsServiceConfig
{
@Getter
private String awsRegion = "";
{
```